### DISCUSSION TITLE
Real life examples with duckdb

### DISCUSSION THREAD

**Discussion by joergklein** [duckdb/duckdb#9578] [Ideas]
The documentation of duckdb is really good. What I am missing are executable examples that can be executed by anyone using copy and paste. I am currently trying to import csv files into duckdb and then process them in `R`.

For example, it could be shown in the example how the implementation is carried out and after the import as an example the calculation of the mean value of a column.

`Import csv files in duckdb and R #9577`

Thank you very much

**Comment by emmenlau** [comment]
I would like to chime in on this idea, with a more specific request.

The first duckdb documentation page at https://duckdb.org/docs/connect is quite helpful. But when it comes to actually persisting data to disk, it becomes very vague. The page just says that typically a client would work on a single file, similar to SQLite, and that the extension is not essential. Now this style of working is exactly what we want. But I could not find any documentation on what file formats are available, how to choose between them, and which ones a beginner would typically use!

SQLite is rather trivial in this aspect, because it (mostly) supports its own db file, no questions asked. How is that with duckdb? In other docs pages, duckdb mentions a large number of import formats, and some of them seem to support writing too (i.e. pandas, ...). Are any of these recommended for beginners duckdb use case?

It would be great if already the front page examples would mention the recommended file format(s), and if there is more than one, highlight a few of the pro's and con's?!

Basically, we want to understand how the following sqlite example translates to duckdb, so what kind of file format we would use:
```
    sqlite3* DB; 
    int status = sqlite3_open("/path/to/database.db", &DB); 
```
